cmd/kubeadm/app/cmd/phases/upgrade/node/kubeletconfig.go

```diff
@@ -71,7 +71,17 @@ func runKubeletConfigPhase() func(c workflow.RunData) error {
 			return err
 		}
 
-		// TODO: Checkpoint the current configuration first so that if something goes wrong it can be recovered
+		// Create a copy of the kubelet config file in the /etc/kubernetes/tmp/ folder.
+		backupDir, err := constants.CreateTempDirForKubeadm(constants.KubernetesDir, "kubeadm-kubelet-config")
+		if err != nil {
+			return err
+		}
+		src := filepath.Join(kubeletDir, constants.KubeletConfigurationFileName)
+		dest := filepath.Join(backupDir, constants.KubeletConfigurationFileName)
+		if err := os.Rename(src, dest); err != nil {
+			return err
+		}
+		fmt.Printf("[upgrade] Kubelet config file is backed up to the %s. \n", backupDir)
```

Comment:
```suggestion
		fmt.Printf("[upgrade] backing up kubelet config file to %s\n", dest)
```

changed it to the full path. maybe use backupDir for consistency with other messages?
also should move it before os.rename, i.e. taking action now

Reply:
> also should move it before os.rename, i.e. taking action now

hmm... I am not following for this, do you mean call some method to move the file to the dest folder before renaming it? if this is the case, I think this change should be fine, rename here does the move action actually.

e.g.
https://github.com/kubernetes/kubernetes/blob/7cba7e6c58078865bedaa3ba7ea30f0c3193ff4d/cmd/kubeadm/app/phases/upgrade/staticpods.go#L125-L127

Reply:
hmm... I missed your comments here, has been fixed.